pkg/dockerregistry/server/auth.go

```diff
@@ -484,3 +500,27 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 	}
 	return nil
 }
+
+func verifyImageSignatureAccess(ctx context.Context, namespace, imageRepo string, client client.LocalSubjectAccessReviewsNamespacer) error {
+	sar := authorizationapi.LocalSubjectAccessReview{
+		Action: authorizationapi.Action{
+			Verb:         "create",
+			Group:        imageapi.GroupName,
+			Resource:     "imagesignatures",
+			ResourceName: imageRepo,
+		},
+	}
+	response, err := client.LocalSubjectAccessReviews(namespace).Create(&sar)
+	if err != nil {
+		context.GetLogger(ctx).Errorf("OpenShift client error: %s", err)
+		if kerrors.IsUnauthorized(err) || kerrors.IsForbidden(err) {
+			return ErrOpenShiftAccessDenied
+		}
+		return err
+	}
+	if !response.Allowed {
+		context.GetLogger(ctx).Errorf("OpenShift access denied: %s", response.Reason)
+		return ErrOpenShiftAccessDenied
+	}
```

Comment:
I know I'm a bit late in review, but that can be fixed as a followup, why we do copy&paste this piece starting from line 513 here and in `verifyImageStreamAccess` and `verifyPruneAccess`, can we squash that to a helper method?